enos/enos-scenario-upgrade.hcl

```diff
@@ -65,7 +65,8 @@ scenario "upgrade" {
     module = module.provision_cluster
     variables {
       name                      = local.cluster_name
-      nomad_local_binary        = step.copy_initial_binary.nomad_local_binary
+      nomad_local_binary        = step.copy_initial_binary.binary_path[matrix.os]
+      nomad_local_binary_server = step.copy_initial_binary.binary_path[local.server_os]
```

Comment:
We should rebase this PR on https://github.com/hashicorp/nomad/pull/25172 once that's been merged.